apps/charge-api-service/src/consensus-api/consensus-api.controller.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  UseGuards,
+  UseInterceptors
+} from '@nestjs/common'
+import { IsValidPublicApiKeyGuard } from '@app/api-service/api-keys/guards/is-valid-public-api-key.guard'
+import { ConsensusApiService } from '@app/api-service/consensus-api/consensus-api.service'
+import { DelegatedAmountsDto } from '@app/network-service/consensus/dto/consensus.dto'
+import { CacheInterceptor } from '@nestjs/cache-manager'
+
+@UseGuards(IsValidPublicApiKeyGuard)
```

Comment:
@LiorAgnin which API key will be used?

Reply:
@MusabShakeel576 Any public API key generated to a project/operator account.

Reply:
@LiorAgnin will we utilize the API key of the user operator account, or pass our own?

Staking user doesn't have to create a project/operator account, so API key in Staking endpoint seems redundant.

Reply:
We don't need to use the connected user's API key. My idea was to create a dedicated API key for the staking DApp.

@MusabShakeel576 @NikolayRn  WDYT?

Reply:
@LiorAgnin if we create our own dedicated API key, utilize it publicly on the Staking Dapp, and also pass it with every call to the Staking endpoint, then it's the same as not using the API key.